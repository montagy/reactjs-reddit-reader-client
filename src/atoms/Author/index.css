.author {
  font-size: 1.4em;
  font-weight: 700;
  color: color(black l(+40%) blue(+20));
  & > span {
    font-size: 0.8em;
    margin-left: 1em;
    color: color(var(--baseColor) b(+60%) alpha(0.8));
  }
}
