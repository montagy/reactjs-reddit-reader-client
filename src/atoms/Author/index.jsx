import React from 'react';
import { number, string, bool } from 'prop-types';
import styles from './index.css';
import { timeAgo } from '../../utils';

Author.propTypes = {
  name: string.isRequired,
  time: number.isRequired,
  isSubmitter: bool,
};
Author.defaultProps = {
  name: '未知',
  time: new Date().getTime(),
};
function Author({ name = '', time, isSubmitter }) {
  const ago = timeAgo(time);
  return (
    <div className={styles.author}>
      {name}
      {isSubmitter && <span>[S]</span>}
      <span>
        发表于{ago.day > 1
          ? `${ago.day}天`
          : ago.hour ? `${ago.hour}小时` : `${ago.min}分钟`}前
      </span>
    </div>
  );
}

export default Author;
