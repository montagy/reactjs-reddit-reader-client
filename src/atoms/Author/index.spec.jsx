import React from 'react';
import renderer from 'react-test-renderer';
import Author from './';

describe('Author', () => {
  it('mount success', () => {
    const comp = renderer
      .create(<Author name="test" time={new Date().getTime()} />)
      .toJSON();
    expect(comp).toMatchSnapshot();
  });
});
