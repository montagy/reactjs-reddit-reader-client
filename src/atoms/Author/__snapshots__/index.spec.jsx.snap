// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author mount success 1`] = `
<div
  className="author"
>
  test
  <span>
    发表于
    0小时
    前
  </span>
</div>
`;
